### PR TITLE
Add types to package.json exports mapping

### PR DESCRIPTION
**Pull request by jhirvioja**
Currently, preact-router errors when in TypeScripts config the moduleResolution is "bundler", "Node16" or "NodeNext". See example of error message here: https://github.com/preactjs/preact-router/issues/443

One proposed fix is to include types in the package.json's exports mapping. See example here: https://github.com/preactjs/preact-router/issues/429

This proposed fix is implemented in this pull request. If this is not a reasonable fix, some other fix should be introduced.

Related: https://github.com/microsoft/TypeScript/issues/52363

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,13 +12,15 @@
 		".": {
 			"module": "./dist/preact-router.mjs",
 			"import": "./dist/preact-router.mjs",
-			"require": "./dist/preact-router.js"
+			"require": "./dist/preact-router.js",
+			"types": "./index.d.ts"
```
[the remark this answers is not in the document]
Comment (rschristian) [reply package.json:16]: Whoops, meant to post this a couple days ago. Sorry about that

[the remark this answers is not in the document]
Comment (jhirvioja) [reply package.json]: Seems so, I did not know that! Added a commit fixing this.

https://www.typescriptlang.org/docs/handbook/esm-node.html
> The "types" condition should always come first in "exports".

[the remark this answers is not in the document]
Comment (rschristian) [reply package.json]: Good to know! Wasn't quite sure myself, haven't kept up with TS in a while.

**Review by rschristian — APPROVED** [review]

**Comment by rschristian** [comment]
Thanks!

**Pull request merged**